### ISSUE TITLE
Tests involving exceptions don't check that exception was thrown

### ISSUE THREAD

**Issue by travisdowns** [eduardoboucas/staticman#327]
Consider [this test](https://github.com/eduardoboucas/staticman/blob/c8e339f65c3e760518a849dbbfb1deb2c8e6410a/test/unit/lib/GitHub.test.js#L112):

~~~js
    test('returns an error if parsing fails for the given file', async () => {
      const filePath = 'path/to/file.yml'
      const githubInstance = await new GitHub(req.params)

      try {
        await githubInstance.readFile(filePath)
      } catch (err) {
        expect(err._smErrorCode).toEqual('PARSING_ERROR')
        expect(err.message).toBeDefined()
      }
    })
~~~

This checks if `readFile` throws the appropriate exception. However, if `readFile` _doesn't throw at all_, it passes!

That's actually happening now, and I ran into it when my "fix" started making this test fail. I am not a javascript developer, but I came up with this idiom to fix this:

~~~js
    test('returns an error if parsing fails for the given file', async () => {
      const filePath = 'path/to/file.yml'
      const githubInstance = await new GitHub(req.params)

      let exception = 'thrown'
      try {
        await githubInstance.readFile(filePath)
        exception = 'not thrown'
      } catch (err) {
        expect(err._smErrorCode).toEqual('PARSING_ERROR')
        expect(err.message).toBeDefined()
      }

      expect(exception).toBe('thrown')
    })
~~~

Does it look reasonable?

**Comment by alexwaibel** [comment]
Good catch! As mentioned in other threads, I'm a bit busy with uni at the moment but I'll definitely take a look at the related PR when I have time. Thanks again for all your contributions @travisdowns

**Comment by travisdowns** [comment]
Withdrawing this since @alexwaibel mentioned this project does not merge changes which cause failures.

**Comment by travisdowns** [comment]
Sorry I was trying to close the PR, but is the issue, not the PR...

**Comment by hispanic** [comment]
The easy way to address this is to insert `expect.hasAssertions()` before the try/catch block. More info: https://jestjs.io/docs/en/expect#expecthasassertions

**Comment by travisdowns** [comment]
> The easy way to address this is to insert `expect.hasAssertions()` before the try/catch block.

Yeah, it's one way, although I don't think it could be a direct drop-in for the existing tests. It only checks that *any* assertion has been executed, but several of the tests have various assertions which will always be attempted, so this wouldn't detect the case where the assertion in the catch was skipped. It works in the example in the documentation where the *only* thing being tested is a single try/catch without any additional assertions. It's also fragile in the sense that if you have any setup code in your test, and someone adds some assertions (possibly in a helper function so not visible in test body), the assumption that the test has only a single assertion is violated.
